test/units/module_utils/facts/system/test_date_time.py

```diff
@@ -0,0 +1,79 @@
+# unit tests for ansible system date_time fact collectors
+# -*- coding: utf-8 -*-
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import datetime
+import time
+
+from units.compat.mock import Mock, patch
+from .. base import BaseFactsTest
+
+from ansible.module_utils.facts.system.date_time import DateTimeFactCollector
+
+
+class MockDatetime(datetime.datetime):
+    @classmethod
+    def now(cls):
+        return cls(2020, 7, 8, 12, 34, 56, 123456)
+
+    @classmethod
+    def utcnow(cls):
+        return cls(2020, 7, 8, 16, 34, 56, 123456)
+
+
+class TestDateTimeFacts(BaseFactsTest):
+    __test__ = True
+    gather_subset = ['!all', 'date_time']
+    valid_subsets = ['date_time']
+    fact_namespace = 'ansible_date_time'
+    collector_class = DateTimeFactCollector
+
+    def _mock_module(self):
+        mock_module = Mock()
+        mock_module.params = {'gather_subset': self.gather_subset,
+                              'gather_timeout': 10,
+                              'filter': '*'}
+        return mock_module
+
+    def test_collect(self):
+        module = self._mock_module()
+        datetime.datetime = MockDatetime
```

Comment:
Mutating global stdlib state like this is a bad idea because it influences the whole interpreter state.
It's common to use `patch` in a form of context manager or a decorator that would ensure to revert the original state of the patched object.

Also, since we use pytest as a testrunner, it's a good idea to just use its built-in [`monkeypatch`](https://docs.pytest.org/en/latest/monkeypatch.html) instead of `patch` from unittest. It will also make sure to restore the state of the patch target upon exiting the test.

Reply:
I agree with this. It would be much better to use a decorator to mock the return value rather than mocking the entire object.